static/js/blueslip.js

```diff
@@ -96,20 +98,102 @@ exports.get_log = function blueslip_get_log() {
     return logger.get_log();
 };
 
-// Format error stacks using the ErrorStackParser
-// external library
-function getErrorStack(stack) {
+var stack_trace_gps = new StackTraceGPS();
+
+function beautify_stackframe(stackframe) {
+    return stack_trace_gps
+        .getMappedLocation(stackframe)
+        .then(function (info) {
+            // If the file is local, just the filename.
+            // Otherwise, show the full path starting from node_modules.
+            var idx = info.fileName.indexOf('node_modules');
+            var full_path, show_path;
+            if (idx !== -1) {
+                full_path = show_path = info.fileName.slice(idx)
+            } else {
+                full_path = info.fileName.split('/').slice(-3).join('/');
+                show_path = full_path.slice(full_path.lastIndexOf('/') + 1);
+            }
+
+            // Don't show the full scope in function name
+            var full_fn_name, fn_name;
+            try {
+                full_fn_name = info.functionName.match(/(?:[a-z_]+\.[jt]s(?:\?\w+)?\.)(.+)/)[1];
+            } catch {
+                full_fn_name = info.functionName;
+            }
+            fn_name = full_fn_name.split('.');
+
+            fn_name[fn_name.length - 1] = '<span class="function-name">'
+                + fn_name[fn_name.length - 1].replace('<', '&lt;')
+                    .replace('>', '&gt;')
+                + '</span>';
+
+            return {
+                full_path: full_path,
+                show_path: show_path,
+                line_number: info.lineNumber,
+                function_name: fn_name.join('.'),
+            };
+        });
+}
+
+function display_stacktrace(error, stack) {
     var ex = new Error();
     ex.stack = stack;
-    return ErrorStackParser
-        .parse(ex)
-        .map(function (stackFrame) {
-            return stackFrame.lineNumber
-            + ': ' + stackFrame.fileName
-            + ' | ' + stackFrame.functionName;
-        }).join('\n');
+    var stacktrace = ErrorStackParser.parse(ex).map(beautify_stackframe);
+
+    // We don't polyfill Promise but since this is available in debug
+    // mode only, assume the developer's browser is recent enough.
+    Promise.all(stacktrace).then(function (stackframes) {
```

Comment:
Maybe expand this comment to explain why it's a promise?  (I assume it's because `beautify_stackframe` has to go over the network to get data?)

Reply:
The stacktrace-gps library returns a promise in all of their APIs. I considered the other stacktrace sourcemapping libraries that doesn't use promises, but their browser support is not great.
* sourcemapped-stacktrace - chrome & firefox
* node-source-map-support - chrome only (primarily for node)

Reply:
Sounds great; let's just expand the comment to explain that (including the reason it does so, which I'm pretty sure is because it might need to fetch the source map from the server)